src/util.ts

```diff
@@ -0,0 +1,191 @@
+    // Sometimes the cell can be in the same depth.
+    const role = $pos.node(d).type.spec.tableRole;
+    if (role === 'cell' || role === 'header_cell') return $pos.node(d);
+  }
+  return null;
+}
+
+/**
+ * @public
+ */
+export function isInTable(state: EditorState): boolean {
+  const $head = state.selection.$head;
+  for (let d = $head.depth; d > 0; d--)
+    if ($head.node(d).type.spec.tableRole == 'row') return true;
+  return false;
+}
+
+/**
+ * @public
+ */
+export function selectionCell(
+  state: EditorState,
+): ResolvedPos | null | undefined {
+  const sel = state.selection as CellSelection | NodeSelection;
+  if ('$anchorCell' in sel && sel.$anchorCell) {
+    return sel.$anchorCell.pos > sel.$headCell.pos
+      ? sel.$anchorCell
+      : sel.$headCell;
+  } else if (
+    'node' in sel &&
+    sel.node &&
+    sel.node.type.spec.tableRole == 'cell'
+  ) {
+    return sel.$anchor;
+  }
+  return cellAround(sel.$head) || cellNear(sel.$head);
+}
+
+function cellNear($pos: ResolvedPos): ResolvedPos | undefined {
+  for (
+    let after = $pos.nodeAfter, pos = $pos.pos;
+    after;
+    after = after.firstChild, pos++
+  ) {
+    const role = after.type.spec.tableRole;
+    if (role == 'cell' || role == 'header_cell') return $pos.doc.resolve(pos);
+  }
+  for (
+    let before = $pos.nodeBefore, pos = $pos.pos;
+    before;
+    before = before.lastChild, pos--
+  ) {
+    const role = before.type.spec.tableRole;
+    if (role == 'cell' || role == 'header_cell')
+      return $pos.doc.resolve(pos - before.nodeSize);
+  }
+}
+
+/**
+ * @public
+ */
+export function pointsAtCell($pos: ResolvedPos): boolean {
+  return $pos.parent.type.spec.tableRole == 'row' && !!$pos.nodeAfter;
+}
+
+/**
+ * @public
+ */
+export function moveCellForward($pos: ResolvedPos): ResolvedPos {
+  return $pos.node(0).resolve($pos.pos + $pos.nodeAfter.nodeSize);
+}
+
+/**
+ * @public
+ */
+export function inSameTable($a: ResolvedPos, $b: ResolvedPos): boolean {
+  return $a.depth == $b.depth && $a.pos >= $b.start(-1) && $a.pos <= $b.end(-1);
+}
+
+/**
+ * @public
+ */
+export function findCell($pos: ResolvedPos): Rect {
+  return TableMap.get($pos.node(-1)).findCell($pos.pos - $pos.start(-1));
+}
+
+/**
+ * @public
+ */
+export function colCount($pos: ResolvedPos): number {
+  return TableMap.get($pos.node(-1)).colCount($pos.pos - $pos.start(-1));
+}
+
+/**
+ * @public
+ */
+export function nextCell(
+  $pos: ResolvedPos,
+  axis: string,
+  dir: number,
+): null | ResolvedPos {
+  const start = $pos.start(-1),
+    map = TableMap.get($pos.node(-1));
+  const moved = map.nextCell($pos.pos - start, axis, dir);
+  return moved == null ? null : $pos.node(0).resolve(start + moved);
+}
+
+/**
+ * @public
+ */
+export function setAttr(
+  attrs: Attrs,
+  name: string,
+  value: unknown,
+): MutableAttrs {
+  const result = {};
+  for (const prop in attrs) result[prop] = attrs[prop];
+  result[name] = value;
+  return result;
+}
```

Comment:
Let's mark this function as `@internal`. We don't need it anymore anyway since we have ES6 object spread `{...attrs, name: value}`.

Reply:
done